assets/blocks/learner-courses-block/learner-courses-edit.js

```diff
@@ -7,17 +7,70 @@ import { omitBy } from 'lodash';
 /**
  * WordPress dependencies
  */
+import { useDispatch } from '@wordpress/data';
+import { createBlock } from '@wordpress/blocks';
 import { useState } from '@wordpress/element';
 import { Icon, image } from '@wordpress/icons';
 import { Button } from '@wordpress/components';
-import { Warning } from '@wordpress/block-editor';
+import { store as blockEditorStore, Warning } from '@wordpress/block-editor';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import ProgressBar from '../../shared/blocks/progress-bar';
 import LearnerCoursesSettings from './learner-courses-settings';
+import { DEFAULT_ATTRIBUTES as COURSE_LIST_DEFAULT_ATTRIBUTES } from '../course-list-block';
+
+/**
+ * Deprecation notice component.
+ *
+ * @param {Object} props
+ * @param {Object} props.clientId Block client ID.
+ */
+const DeprecationNotice = ( { clientId } ) => {
```

Comment:
I decided not to write tests here since it's more about editor integration and it would probably need e2e tests to make sense.